libraryValidations/azuredeploy.json

```diff
@@ -0,0 +1,125 @@
+      "type": "Microsoft.OperationalInsights/workspaces",
+      "apiVersion": "2023-09-01",
+      "name": "[parameters('workspaceName')]",
+      "location": "[parameters('location')]",
+      "properties": {}
+    },
+    {
+      "type": "Microsoft.Insights/components",
+      "apiVersion": "2020-02-02",
+      "name": "[parameters('appInsightsName')]",
+      "location": "[parameters('location')]",
+      "kind": "web",
+      "properties": {
+        "Application_Type": "web",
+        "WorkspaceResourceId": "[resourceId('Microsoft.OperationalInsights/workspaces', parameters('workspaceName'))]"
+      },
+      "dependsOn": [
+        "[resourceId('Microsoft.OperationalInsights/workspaces', parameters('workspaceName'))]"
+      ]
+    },
+    {
+      "type": "Microsoft.AppConfiguration/configurationStores",
+      "apiVersion": "2024-06-01-preview",
+      "name": "[parameters('appConfigName')]",
+      "location": "[parameters('location')]",
+      "sku": {
+        "name": "Standard"
+      },
+      "properties": {
+        "publicNetworkAccess": "Enabled",
+        "telemetry": {
+          "encryption": {},
+          "disableLocalAuth": false,
+          "softDeleteRetentionInDays": 7,
+          "enablePurgeProtection": false,
+          "dataPlaneProxy": {
+            "authenticationMode": "Local",
+            "privateLinkDelegation": "Disabled"
+          },
+          "resourceId": "[resourceId('Microsoft.Insights/components', parameters('appInsightsName'))]"
+        }
+      },
+      "dependsOn": [
+        "[resourceId('Microsoft.Insights/components', parameters('appInsightsName'))]"
+      ]
+    },
+    {
+      "type": "Microsoft.AppConfiguration/configurationStores/keyValues",
+      "apiVersion": "2021-03-01-preview",
+      "name": "[concat(parameters('appConfigName'), '/','.appconfig.featureflag~2FComplete')]",
+      "dependsOn": [
+        "[resourceId('Microsoft.AppConfiguration/configurationStores', parameters('appConfigName'))]"
+      ],
+      "properties": {
+        "value": "{\"id\":\"Complete\",\"description\":\"\",\"enabled\":true,\"conditions\":{\"client_filters\":[{\"name\":\"Microsoft.TimeWindow\",\"parameters\":{\"Start\":\"Mon, 21 Oct 2024 07:00:00 GMT\",\"End\":\"Fri, 22 Oct 3024 07:00:00 GMT\"}}]},\"variants\":[{\"name\":\"Large\",\"configuration_value\":100},{\"name\":\"Medium\",\"configuration_value\":50},{\"name\":\"Small\",\"configuration_value\":10}],\"allocation\":{\"percentile\":[{\"variant\":\"Large\",\"from\":0,\"to\":25},{\"variant\":\"Medium\",\"from\":25,\"to\":55},{\"variant\":\"Small\",\"from\":55,\"to\":95},{\"variant\":\"Large\",\"from\":95,\"to\":100}],\"group\":[{\"variant\":\"Large\",\"groups\":[\"beta\"]}],\"user\":[{\"variant\":\"Small\",\"users\":[\"Rachel\"]}],\"seed\":\"test-seed\",\"default_when_enabled\":\"Medium\",\"default_when_disabled\":\"Medium\"},\"telemetry\":{\"enabled\":true}}",
+        "contentType": "application/vnd.microsoft.appconfig.ff+json;charset=utf-8"
```

Comment:
I wonder if we should instead have these feature flags as a json in `Samples` folder. This way someone can easily change them and change the expected result of the tests as well. The inputs and expected results should be collocated IMO.

What do you think?

Reply:
I'm rather unfamiliar with arm templates/azd if we go that way. Can we deploy json configs from another file? This is the way I've always done it.

Reply:
I don't think so, but after the deployment, we can use Powershell to push these key values

Reply:
We will do an import/export in the future. For now, we will stay with this